test/integration/kernel.spec.ts

```diff
@@ -8505,3 +8301,127 @@ describe('Kernel', () => {
+			toVersionedType: 'org@1.0.0',
+			name: 'is member of',
+			inverseName: 'has member',
+			title: 'User',
+			inverseTitle: 'Organization',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'org@1.0.0',
+			name: 'is part of',
+			inverseName: 'has member',
+			title: 'Card',
+			inverseTitle: 'Organization',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'card@1.0.0',
+			name: 'is attached to',
+			inverseName: 'has attached element',
+			title: 'Attachment',
+			inverseTitle: 'Attachment',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'card@1.0.0',
+			name: 'is linked to',
+			inverseName: 'has link',
+			title: 'Link',
+			inverseTitle: 'Link',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'card@1.0.0',
+			name: 'is child of',
+			inverseName: 'has child', // It is defined like this in tests above
+			title: 'Card',
+			inverseTitle: 'Card',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'card@1.0.0',
+			name: 'is appended to',
+			inverseName: 'has appended element',
+			title: 'Card',
+			inverseTitle: 'Card',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'card@1.0.0',
+			name: 'is owned by',
+			inverseName: 'owns',
+			title: 'Card',
+			inverseTitle: 'Owner',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'card@1.0.0',
+			name: 'works at',
+			inverseName: 'has worker',
+			title: 'Worker',
+			inverseTitle: 'Workplace',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'card@1.0.0',
+			name: 'believes in',
+			inverseName: 'is believed by',
+			title: 'Believer',
+			inverseTitle: 'Deity',
+		},
+		{
+			fromVersionedType: 'card@1.0.0',
+			toVersionedType: 'card@1.0.0',
+			name: 'reports to',
+			inverseName: 'receives reports from',
+			title: 'Reporter',
+			inverseTitle: 'Manager',
+		},
+	];
+
+	const relContracts = relSpecs.map((spec) => {
+		const fromType = spec.fromVersionedType.split('@')[0];
+		const toType = spec.toVersionedType.split('@')[0];
+		const dashedName = _.kebabCase(spec.name);
+		return {
+			slug: `relationship-${fromType}-${dashedName}-${toType}`,
+			type: 'relationship@1.0.0',
+			name: spec.name,
+			data: {
+				inverseName: spec.inverseName,
+				title: spec.title,
+				inverseTitle: spec.inverseTitle,
+				from: {
+					type: spec.fromVersionedType,
+				},
+				to: {
+					type: spec.toVersionedType,
+				},
+			},
+		};
+	});
+
+	const createdContracts = await Promise.all(
+		relContracts.map((rc) =>
+			ctx.kernel.insertContract(ctx.logContext, ctx.kernel.adminSession()!, rc),
+		),
+	);
+
+	createdContracts.forEach((cc) => assert(cc !== null));
+
+	// wait until all relationships are loaded into the kernel
+	let allLoaded = false;
+	do {
+		const loadedRelationships = ctx.kernel.getRelationships();
+		allLoaded = loadedRelationships.length >= createdContracts.length;
+		await Bluebird.delay(1000);
+	} while (!allLoaded);
```

Comment:
You could instead use the `retry()` test utility for this kind of "wait until" work: https://github.com/product-os/autumndb/blob/master/lib/test-utils.ts#L167

Reply:
Thanks for pointing this; and also for adding the integration tests on the `relationships` cache.
Updating the code